src/state/middleware/scigateway.middleware.tsx

```diff
@@ -270,9 +286,25 @@ export const autoLoginMiddleware: Middleware<
         action.type === InvalidateTokenType)
     ) {
       next(action);
-      return await autoLogin().then(() => {
-        dispatch(autoLoginAuthorised());
-      });
+      return await autoLogin()
+        .then(() => {
+          dispatch(autoLoginAuthorised());
+        })
+        .catch(() => {
+          log.error('Auto Login via middleware failed');
+          // we can't recover from here - tell user they'll need to refresh the page or login again
+          document.dispatchEvent(
+            new CustomEvent(microFrontendMessageId, {
+              detail: {
+                type: NotificationType,
+                payload: {
+                  severity: 'error',
+                  message: getString(res, 'auto-login-error-msg'),
```

Comment:
Looks good from this side but I think you forgot to define the `auto-login-error-msg`. I was going to make a commit to quickly add it myself but wanted your thoughts on what the contents of this message should actually be. This would be a fairly unique situation where the auto login has failed for some reason. Is it the best idea to ask the user to "reload the page" and try again or should we advise them to do something else entirely like try logging in for real. Because if the auto login fails, there could be a bigger problem in the backend as a whole, I'm not sure.

Reply:
Good catch! I have actually wrote the message, just must have missed committing it. I'll add it myself soon so you don't need to worry about the wording. I did write a message along the lines of `please refresh the page and if the error persists contact support` or something